python/adelphi/setup.py

```diff
@@ -1,5 +1,16 @@
 from setuptools import setup
 
+PY3 = sys.version_info[0] == 3
+
+dependencies = [
+    'cassandra-driver >= 3.24.0',
```

Comment:
I believe this should remain a static/pinned version instead of a lower bound.
We don't want to suddenly pick up a newer version that exposes more metadata that's not covered by the anonimyzer.

Reply:
I think this definitely could use another pass.

The original intent here was to pin deps to the lowest supported version for Python 2.7 while allowing them to go higher for Py3k.  My thinking was that projects would have a final version that ran on Python 2.7; PyGithub certainly does this.  So I wanted to make sure that baseline was supported while allowing Py3k instances to use higher versions.

This was my original attempt at dep management; it came before the subsequent logic which introduces an explicit check for Py3k + changes to deps in case it finds it.  That would probably be a cleaner way to handle this now.

Interestingly, it's also worth noting that PyGithub might be an exception.  The cassandra-driver package works with both 2.7 and 3.x so it could be explicitly pinned.  The same is true for click; both 2.7 and 3.x use 7.1.2.

There's _also_ the point that even if new metadata is introduced that's actually okay as long as they maintain backwards compatibility with the current version.  The better way to express _that_ idea would be to use something like Ruby's ~> comparison (at least as long as the package supports semantic versioning).  Moving to something like that may be worthwhile too.

So yeah, in short... I wanna take another look at this. :(